File: Java/EstruturaDeRepeticao/Exercicio27.java
// Faça um programa que calcule o número médio de alunos por turma.
// Para isto, peça a quantidade de turmas e a quantidade de alunos para cada turma.
// As turmas não podem ter mais de 40 alunos.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio27 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite a quantidade de turmas: ");
        int quantidadeTurmas = scanner.nextInt();

        int totalAlunos = 0;

        for (int i = 1; i <= quantidadeTurmas; i++) {
            int alunosTurma;
            do {
                System.out.print("Digite a quantidade de alunos da turma " + i + " (não pode ser maior que 40): ");
                alunosTurma = scanner.nextInt();
            } while (alunosTurma <= 0 || alunosTurma > 40);

            totalAlunos += alunosTurma;
        }

        double mediaAlunosPorTurma = (double) totalAlunos / quantidadeTurmas;

        System.out.println("Número médio de alunos por turma: " + mediaAlunosPorTurma);

        scanner.close();
    }
}

File: Python/EstruturaDeRepeticao/exercicio_27.py
"""
Faça um programa que calcule o número médio de alunos por turma.
Para isto, peça a quantidade de turmas e a quantidade de alunos para cada turma.
As turmas não podem ter mais de 40 alunos.
"""

quantidade_turmas = int(input("Digite a quantidade de turmas: "))

total_alunos = 0
for i in range(1, quantidade_turmas + 1):
    quantidade_alunos = int(input(f"Digite a quantidade de alunos na turma {i} (Menos de 40): "))
    if quantidade_alunos > 40:
        print("Comando inválido!")
        exit()
    total_alunos += quantidade_alunos

media_alunos_por_turma = total_alunos / quantidade_turmas

print(f"A média de alunos por turma é: {media_alunos_por_turma:.1f}")
